buzsaki_lab_to_nwb/tingley_code_septal/tingleyseptalbehaviorinterface.py

```diff
@@ -0,0 +1,167 @@
+"""Authors: Heberto Mayorquin and Cody Baker."""
+from pathlib import Path
+import warnings
+
+import numpy as np
+from hdmf.backends.hdf5.h5_utils import H5DataIO
+
+from pynwb.file import NWBFile, TimeIntervals
+from pynwb.behavior import SpatialSeries, Position, CompassDirection
+from nwb_conversion_tools.basedatainterface import BaseDataInterface
+from nwb_conversion_tools.utils.conversion_tools import get_module
+from nwb_conversion_tools.utils.json_schema import FolderPathType
+
+from .tingleyseptal_utils import read_matlab_file
+
+
+class TingleySeptalBehaviorInterface(BaseDataInterface):
+    """Behavior data interface for the Tingley Septal project."""
+
+    def __init__(self, folder_path: FolderPathType):
+        super().__init__(folder_path=folder_path)
+
+    def run_conversion(self, nwbfile: NWBFile, metadata: dict):
+        session_path = Path(self.source_data["folder_path"])
+        session_id = session_path.stem
+
+        # Load the file with behavioral data
+        behavior_file_path = Path(session_path) / f"{session_id}.behavior.mat"
+        behavior_mat = read_matlab_file(str(behavior_file_path))["behavior"]
+
+        # Add trials
+        events = behavior_mat["events"]
+        trial_interval_list = events["trialIntervals"]
+
+        data = []
+        for start_time, stop_time in trial_interval_list:
+            data.append(
+                dict(
+                    start_time=float(start_time),
+                    stop_time=float(stop_time),
+                )
+            )
+        [nwbfile.add_trial(**row) for row in sorted(data, key=lambda x: x["start_time"])]
+
+        trial_list = events["trials"]
+        direction_list = [trial.get("direction", "") for trial in trial_list]
+        trial_type_list = [trial.get("type", "") for trial in trial_list]
+
+        if not all([direction == "" for direction in direction_list]):
+            nwbfile.add_trial_column(name="direction", description="direction of the trial", data=direction_list)
+
+        if not all([trial_type == "" for trial_type in trial_type_list]):
+            nwbfile.add_trial_column(name="trial_type", description="type of trial", data=trial_type_list)
+
+        # Position
+        module_name = "Position"
```

Comment:
```suggestion
        module_name = "position"
```

Reply:
can we try to stick with the pynwb modules for module names? e.g. for this can be put it in "behavior"? See best practices [here](https://www.nwb.org/best-practices/#naming-of-processing-modules)

Reply:
That'd be fine; would `neural_states` count as `behavior` or `ecephys` (since they are inferred from the `ecephys` directly)

Reply:
I would say ecephys